src/Modules/Manage/Dnn.PersonaBar.Users/Components/Prompt/Commands/DeleteUser.cs

```diff
@@ -56,21 +58,18 @@ public override ConsoleResultModel Run()
                 return new ConsoleErrorResultModel(LocalizeString("Prompt_UserAlreadyDeleted"));
             }
 
-            if (!UserController.DeleteUser(ref userInfo, Notify, false))
+            if (!_userControllerWrapper.DeleteUserAndClearCache(ref userInfo, Notify, false))
             {
                 return new ConsoleErrorResultModel(LocalizeString("Prompt_UserDeletionFailed"))
                 {
                     Data = userModels
                 };
             }
-            // We must clear User cache or else, when the user is 'removed' (so it can't be restored), you 
-            // will not be able to create a new user with the same username -- even though no user with that username
-            // exists.
-            DotNetNuke.Common.Utilities.DataCache.ClearUserCache(PortalId, userInfo.Username);
-            // attempt to retrieve the user from the dB
-            userInfo = UserController.GetUserById(PortalId, userInfo.UserID);
+
+            // attempt to retrieve the user from the dB 
+            userInfo = _userControllerWrapper.GetUserById(userInfo.PortalID, userInfo.UserID);
```

Comment:
That said, isn't the purpose of `ref UserInfo` parameter that it shouldn't be necessary to read again? I'd expect it to be returned in its modified state already.

Reply:
So, I'm going to put portalId in local variable and then used in above call

Reply:
Just found out this method call is changing portalId with MasterId of SiteGroup. So local variable soln. would be save